test/mock.test.ts

```
@@ -0,0 +1,62 @@
+ * This script verifies the syntax and structure of the Redis client
+ * without actually connecting to a Redis server.
+ */
+
+// We need to mock the unix-dgram module before importing Redis
+// In TypeScript, we can use jest.mock or a similar approach
+// For this simple test, we'll create a mock module and use dynamic imports
+
+// Define the mock for unix-dgram
+const mockUnixDgram = {
+  createSocket: (type: string) => ({
+    on: (event: string, callback: Function) => mockSocket,
```

Biome (1.9.4)

[error] Don't use 'Function' as a type.

Prefer explicitly define the function shape. This type accepts any function-like value, which can be a common source of bugs.

(lint/complexity/noBannedTypes)